File: src/main/java/com/bisoshi/karta/modules/document/model/Document.java
package com.bisoshi.karta.modules.document.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "documents")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Document {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @Column(name = "project_id", nullable = false)
    private UUID projectId;
    
    @Column(nullable = false)
    private String name;
    
    @Column(name = "file_path_storage", nullable = false)
    private String filePathStorage;
    
    @Column(name = "file_size")
    private Long fileSize;
    
    @Column(name = "mime_type")
    private String mimeType;
    
    @Column(name = "uploaded_by", nullable = false)
    private UUID uploadedBy;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority = Priority.MEDIA;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/document.ts
export enum Priority {
  ALTA = 'ALTA',
  MEDIA = 'MEDIA',
  BAJA = 'BAJA',
}

export interface Document {
  id: string;
  projectId: string;
  name: string;
  filePathStorage: string;
  fileSize: number;
  mimeType: string;
  uploadedBy: string;
  uploadedByName?: string;
  createdAt: string;
  priority: Priority;
}

export interface DocumentResponse {
  id: string;
  projectId: string;
  name: string;
  description?: string;
  fileSize: number;
  mimeType: string;
  uploadedBy: string;
  uploadedByName?: string;
  createdAt: string;
  priority: Priority;
}

export interface UploadDocumentRequest {
  file: File;
  name: string;
  priority: Priority;
  description?: string;
}

export interface UpdateDocumentRequest {
  name?: string;
  priority?: Priority;
  description?: string;
}
